File: L3/python/nta.py
import sys

def nta(n):
    flag = False
    arr = []
    for i in range(0, n + 2):
        arr.append(0)
    arr[1] = 2
    arr[2] = 3
    licz = 5
    k = 2
    while k < n:
        flag = True
        for i in range(1, k + 1):
            if arr[i] * arr[i] > licz:
                break
            if licz % arr[i] == 0:
                flag = False
                break

        if flag:
            k = k + 1
            arr[k] = licz
        licz = licz + 2
    return arr[n]



def main():
    if len(sys.argv) != 2:
        print("Zła liczba argumentów")
        return
    n = int(sys.argv[1])
    if n < 0:
        print("Liczby nie mogą być mniejsze od 0")
        return
    print(nta(n))
    


if __name__ == "__main__":
    main()

File: L3/c/nta.c
#include <stdio.h>
#include <stdlib.h>
#include <stdint.h>
#include <stdbool.h>
#include <math.h>

uint32_t nta(uint32_t n){

    uint32_t* arr = malloc(sizeof(uint32_t) * (n + 1));
    bool flag = false;
    arr[1] = 2;
    arr[2] = 3;
    uint32_t licz = 5, k = 2;

    while (k < n) {
        flag = true;
        for (uint32_t i = 1; i <= k; i++){
            if (arr[i] > floor(sqrt(licz))) break;
            if (licz % arr[i] == 0) {
                flag = false;
                break;
            }
        }
        if (flag) {
            k++;
            arr[k] = licz;
        }
        licz += 2;
    }


    for(uint32_t i = 1; i <= n; i++){
        printf("arr[%u] = %u\n", i, arr[i]);
    }

    uint32_t ret = arr[n];

    free(arr);
    return ret;
}

int main(int argc, char** argv){
    nta(100);
    return 0;
}
